samples/getting_started/CatStatesAndGHZ.qs

```diff
@@ -0,0 +1,63 @@
+import Std.Diagnostics.*;
+
+/// # Summary
+/// Greenberger–Horne–Zeilinger and Cat state sample
+///
+/// # Description
+/// This Q# program shows how to prepare the GHZ state in a register of 3 qubits
+/// and a generalized GHZ state (Cat state) in a register of 5 qubits.
+///
+/// # Remarks
+/// The Greenberger–Horne–Zeilinger state, or GHZ state, is a state with 3
+/// qubits defined as: |GHZ〉 = (|000〉 + |111〉) / √2.
+///
+/// The GHZ state is said to be a maximally entangled state, a multi-qubit
+/// state where the state of any one qubit is not separable from the state
+/// of any of the other qubits.
+///
+/// The generalized form of the GHZ state across any number of qubits is
+/// called a Cat state, and the GHZ state is a special case of the Cat
+/// state where the number of qubits is 3.
```

Comment:
This isn't really true. The Bell states are specifically among 2 qubits, and the GHZ state is among 3 or more states. So "cat state" isn't really a generalized form of GHZ, just another name for it. As long as the register has more than two qubits in a superposition of exactly two states it is accurate to call it a cat state or a GHZ state.

Reply:
I'm not really sure I understand how the GHZ and the Bell states are related. The GHZ is a superposition between the "All" or "Nothing" basis states, but only one of the 4 Bell states is like that. Bell states, to me, look a lot more like the collection of states in superposition between states of the same parity, where parity is the "even" or "odd" count of |1⟩'s. Thinking of them like that, Bell states _could_ be generalizable to any number of qubits, but that might be called something else, and the GHZ would not qualify under that as it is a superposition between an "even" state and ad "odd" state for all odd numbers of qubits.

Reply:
Updated with new comments. Check it out one more time.